src/index.ts

```diff
@@ -121,16 +145,21 @@ export class CSSStyleObserver {
   /**
    * Collect CSS variable values and invoke callback.
    */
-  private _handleUpdate(): void {
+  private _handleUpdate(event?: TransitionEvent): void {
     if (this._targetElement) {
       const computedStyle = getComputedStyle(this._targetElement);
-
       const variables: CSSDeclarations = {};
 
-      this._observedVariables
-        .forEach(value => {
+      if (this._callbackMode === CallbackMode.INDIVIDUAL && event?.propertyName) {
+        const changedProperty = event.propertyName;
+        if (this._observedVariables.includes(changedProperty)) {
```

Comment:
I think this check – that checks if the `event.propertyName` is part of the observed variables – can apply to both modes.

Suggestion to move the check up a level and have it return early.

Something like this:
- If property is given but the property is not observed: do an early return
- If callback mode is individual and property is given: only get individual property
- If callback mode is all: get all that are observed

Reply:
I’ve refactored the code to use a strategy pattern for handling callback modes, moving the modeHandlers outside of the _handleUpdate function (it can have multiple modes in the future). I kept the `?? CallbackMode.INDIVIDUAL` in place to ensure that, in non-type-safe environments, any invalid or garbage values for callbackMode still default to a safe, predictable behavior